bazel/external/http_parser/http_parser.c

```diff
@@ -427,6 +427,7 @@ enum http_host_state
 #define IS_USERINFO_CHAR(c) (IS_ALPHANUM(c) || IS_MARK(c) || (c) == '%' || \
   (c) == ';' || (c) == ':' || (c) == '&' || (c) == '=' || (c) == '+' || \
   (c) == '$' || (c) == ',')
+#define IS_SCHEMA_CHAR(c)   (IS_ALPHANUM(c) || (c) == '.' || (c) == '-' || (c) == '+')
```

Comment:
This is standard compliant. However Envoy limits scheme to `http` or `https` when checking the header map. https://github.com/envoyproxy/envoy/blob/073ed39fc86b64c00b28439d7579a03c19608a8f/source/common/http/header_utility.cc#L188

Reply:
The scheme check is only done for:
* request url: https://github.com/envoyproxy/envoy/blob/eb352bb05d37a8ad4b6cc7012dc7fef74767b2ad/source/common/http/http1/codec_impl.cc#L1100
* x-forward-proto https://github.com/envoyproxy/envoy/blob/33bb6da5cf52f001d0bc32570ea4f9ca211e52a9/source/common/router/router.cc#L88 https://github.com/envoyproxy/envoy/blob/a4053922d7311e11c18e764ff5e5bfa8f71cd334/source/common/http/conn_manager_utility.cc#L32
* scheme https://github.com/envoyproxy/envoy/blob/1bea8aea1590ffab5bb185d32232f9bf265e6b1e/source/extensions/filters/common/mutation_rules/mutation_rules.cc#L111

However the http parser is used for validating URLs for other header values, e.g. referer, without restricted the scheme:
https://github.com/envoyproxy/envoy/blob/613aa3329c774787cbee18725a2bacd2f863637c/source/common/http/conn_manager_utility.cc#L110-L117
https://github.com/envoyproxy/envoy/blob/b67a3fc250cd9f74fde744d400d7c6857ea80dea/source/common/http/utility.cc#L332-L372.

The fix (see linked issue) is for referer header, and not the actual request url.